### PR TITLE
Fix errors in php 7.4 (probably the first batch)

### PR DESCRIPTION
**Pull request by tatankat**

### CODE REVIEW

File: core/framework/Settings.php [hunk 1]
```diff
@@ -297,7 +297,7 @@ public static function get($name, $module = 'core', $scope = null, $uid = 0)
             if ($scope != Context::getScope()->getID() && $scope !== null)
             {
                 $setting = self::_loadSetting($name, $module, $scope);
-                return $setting[$uid];
+                return ($setting !== null?$setting[$uid]:null);
```
Comment (zegenie) [inline core/framework/Settings.php:300]: you can use `return $setting[$uid] ?? null` here instead.

File: core/modules/main/templates/_comments.inc.php [hunk 2]
```diff
@@ -56,5 +56,5 @@
     <div class="initial-placeholder" <?php if ($comment_count == 0): ?>style="display: none;"<?php endif; ?>><span><?= fa_image_tag('check-circle', ['class' => 'icon'], 'far'); ?><span><?= __('Issue created'); ?></span></span></div>
 <?php endif; ?>
 <div id="comments_box">
-    <?php include_component('main/commentlist', compact('comment_count_div', 'mentionable_target_type', 'target_type', 'target_id', 'issue')); ?>
+    <?php include_component('main/commentlist', compact('comment_count_div', 'mentionable_target_type', 'target_type', 'target_id', isset($issue)?'issue':array())); ?>
```
Comment (zegenie) [inline core/modules/main/templates/_comments.inc.php:59]: it's cleaner to add `if (!isset($this->issue)) $this->issue = null;` in the method `componentComments()`

File: core/modules/main/templates/_issue_workflow_transition.inc.php [hunk 3]
```diff
@@ -2,7 +2,7 @@
     <?php if ($issue->isWorkflowTransitionsAvailable()): ?>
         <?php foreach ($issue->getAvailableWorkflowTransitions() as $transition): ?>
             <?php if ($transition instanceof \thebuggenie\core\entities\WorkflowTransition && $transition->hasTemplate()): ?>
-                <?php include_component($transition->getTemplate(), compact('issue', 'transition', 'ajax')); ?>
+                <?php include_component($transition->getTemplate(), compact('issue', 'transition', isset($ajax)?'ajax':array())); ?>
```
Comment (zegenie) [inline core/modules/main/templates/_issue_workflow_transition.inc.php:5]: it's preferrable to initialize with empty values in the associated `componentX()` method, so you don't have to do these checks everywhere. 
```
if (!isset($this->ajax)) $this->ajax = null;
```
in the component method should do it.

File: core/modules/main/templates/_reportissuecontainer.inc.php [hunk 4]
```diff
@@ -4,6 +4,6 @@
         <a href="javascript:void(0);" class="closer" onclick="TBG.Main.Helpers.Backdrop.reset();"><?= fa_image_tag('times'); ?></a>
     </div>
     <div id="backdrop_detail_content" class="backdrop_detail_content">
-        <?php include_component('main/reportissue', compact('selected_project', 'issue', 'issuetypes', 'selected_issuetype', 'selected_milestone', 'selected_build', 'parent_issue', 'errors', 'permission_errors', 'board', 'locked_issuetype')); ?>
+        <?php include_component('main/reportissue', compact('selected_project', isset($issue)?'issue':array(), 'issuetypes', 'selected_issuetype', 'selected_milestone', 'selected_build', 'parent_issue', 'errors', isset($permission_errors)?'permission_errors':array(), 'board', 'locked_issuetype')); ?>
```
Comment (zegenie) [inline core/modules/main/templates/_reportissuecontainer.inc.php:7]: See previous comments about initializing the methods in the associated component method

File: core/modules/main/templates/login.html.php [hunk 5]
```diff
@@ -6,7 +6,7 @@
 <div id="login_backdrop">
     <div class="backdrop_box login_page login_popup" id="login_popup">
         <div id="backdrop_detail_content" class="backdrop_detail_content rounded_top login_content">
-            <?php include_component('main/login', compact('section', 'error')); ?>
+            <?php include_component('main/login', compact('section', isset($error)?'error':array())); ?>
```
Comment (zegenie) [inline core/modules/main/templates/login.html.php:9]: See previous comments about initializing the methods in the associated component method

File: modules/publish/templates/_leftmenu.inc.php [hunk 6]
```diff
@@ -1,6 +1,6 @@
 <?php include_component('main/menulinks', array('links' => $links, 'target_type' => 'wiki', 'target_id' => $links_target_id, 'title' => __('Wiki menu'))); ?>
 <?php if ($article instanceof \thebuggenie\modules\publish\entities\Article && $article->getID()): ?>
     <?php include_component('publish/whatlinkshere', compact('article')); ?>
-    <?php include_component('publish/tools', compact('special', 'article')); ?>
+    <?php include_component('publish/tools', compact(isset($special)?'special':array(), 'article')); ?>
```
Comment (zegenie) [inline modules/publish/templates/_leftmenu.inc.php:4]: See previous comments about initializing the methods in the associated component method

**Review by zegenie — CHANGES_REQUESTED** [review]
Thanks for taking the time to post this. Could you have a look at the suggestions and let me know your thoughts?

**Comment by tatankat** [comment]
It's indeed cleaner that way (I wouldn't have found out, even if I was not doing some quick fixes while waiting for pachno ;) )
I cleaned up the commits and simply removed the magic quotes code as TBG does not support php 7 anymore.

**Review by zegenie — APPROVED** [review]

**Pull request merged**